From ef4d80bbcb32dcad828d97e47223c3f5b453ae5d Mon Sep 17 00:00:00 2001
From: Angela Czubak <aczubak@marvell.com>
Date: Thu, 23 May 2019 16:36:06 +0200
Subject: [PATCH 252/386] octeontx: fix pki_port_open

Loopback does not use packet instruction header by default. If
instruction header is enabled then PKI parsing returns gibberish if
there is no such header.

Change-Id: Ica5a3154dbb685b8a35ee0609feb5bac5212482b
Signed-off-by: Angela Czubak <aczubak@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/9826
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c b/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c
index d25f4e9ffe13..37d4abebe436 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/pki_config.c
@@ -262,7 +262,7 @@ int pki_port_open(struct pkipf_vf *vf, u16 vf_id,
 	for (i = 0; i < pki->max_cls; i++)
 		pki_reg_write(pki, PKI_CLX_PKINDX_STYLE(i, port->pkind), cfg);
 
-	if (port_data->port_type == OCTTX_PORT_TYPE_NET) {
+	if (port_data->port_type != OCTTX_PORT_TYPE_PCI) {
 		cfg = port->has_fcs ? (0x1ULL << PKI_PKIND_CFG_FCS_SHIFT) : 0;
 		for (i = 0; i < pki->max_cls; i++)
 			pki_reg_write(pki, PKI_CLX_PKINDX_CFG(i, port->pkind),
-- 
2.17.1

